tokio/src/runtime/io/driver.rs

```diff
@@ -236,6 +236,54 @@ impl Handle {
         Ok(scheduled_io)
     }
 
+    /// Registers an I/O resource with the reactor, bypassing Mio entirely and using raw epoll_ctl.
+    ///
+    /// This is more or less the copy-pasted from the Mio code, and exists so that we can use flags
+    /// other than the base set of epoll ones we normally use and those representing interests.
+    ///
+    /// This is important for supporting things like `EPOLLEXCLUSIVE`, which is very useful for
+    /// shared-nothing runtimes.
+    ///
+    /// The registries token is returned.
+    #[cfg(all(target_os = "linux", tokio_unstable))]
+    pub(super) fn add_source_raw(
+        &self,
+        source: &mut impl std::os::unix::io::AsRawFd,
+        flags: u32,
+    ) -> io::Result<Arc<ScheduledIo>> {
+        use libc::EPOLLET;
+        use std::os::unix::io::AsRawFd;
+
+        let events = EPOLLET as u32 | flags;
+
+        let scheduled_io = self.registrations.allocate(&mut self.synced.lock())?;
+        let token = scheduled_io.token();
+
+        let mut event = libc::epoll_event {
+            events,
+            u64: usize::from(token) as u64,
+        };
+
+        // TODO: if this returns an err, the `ScheduledIo` leaks...
+        let res = unsafe {
+            libc::epoll_ctl(
```

Comment:
We can't call `epoll_*` methods from Tokio as there is no guarantee that Mio uses epoll under the hood. Either we should have mio expose more APIs and call those, or we need to use epoll directly in Tokio (obviously, the first option is preferable).

Reply:
What guarantees are made by Mio around it's fd?

Reply:
Which also I suppose begs the question "if it isn't here so people can do this as an escape hatch, why is it here at all?"

Reply:
That being said, there's a deeper question here, which is "do we need Interest::CUSTOM() in Mio, and how would that work?"

Personally, I think if we're going to continue to use Mio within tokio, we probably need something like this.